validator/client/beacon-api/propose_beacon_block.go

```diff
@@ -62,7 +64,42 @@ func (c *beaconApiValidatorClient) proposeBeaconBlock(ctx context.Context, in *e
 	}
 
 	headers := map[string]string{"Eth-Consensus-Version": res.consensusVersion}
-	err = c.jsonRestHandler.Post(ctx, endpoint, headers, bytes.NewBuffer(res.marshalledJSON), nil)
+
+	// Try PostSSZ first with SSZ data
+	if res.marshalledSSZ != nil {
+		headers["Content-Type"] = "application/octet-stream"
+		_, _, err = c.jsonRestHandler.PostSSZ(ctx, endpoint, headers, bytes.NewBuffer(res.marshalledSSZ))
+		if err != nil {
+			errJson := &httputil.DefaultJsonError{}
+			// If PostSSZ fails with 406 (Not Acceptable), fall back to JSON
+			if errors.As(err, &errJson) && errJson.Code == http.StatusNotAcceptable && res.marshalJSON != nil {
+				log.WithError(err).Warn("Failed to submit block ssz, falling back to JSON")
+				jsonData, jsonErr := res.marshalJSON()
+				if jsonErr != nil {
+					return nil, errors.Wrap(jsonErr, "failed to marshal JSON fallback")
+				}
+				// Reset headers for JSON
+				headers = map[string]string{"Eth-Consensus-Version": res.consensusVersion}
+				err = c.jsonRestHandler.Post(ctx, endpoint, headers, bytes.NewBuffer(jsonData), nil)
+				// If JSON also fails, return that error; otherwise continue
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to submit block via JSON fallback")
+				}
+			} else {
+				// For non-406 errors or when no JSON fallback is available, return the SSZ error
+				return nil, errors.Wrap(err, "failed to submit block ssz")
+			}
+		}
+	} else if res.marshalJSON != nil {
+		// No SSZ data available, marshal and use JSON
+		jsonData, jsonErr := res.marshalJSON()
+		if jsonErr != nil {
+			return nil, errors.Wrap(jsonErr, "failed to marshal JSON")
+		}
+		err = c.jsonRestHandler.Post(ctx, endpoint, headers, bytes.NewBuffer(jsonData), nil)
+	} else {
+		return nil, errors.New("no marshalling functions available")
+	}
```

Comment:
```suggestion
	} else if res.marshalJSON == nil {
		return nil, errors.New("no marshalling functions available")
	}
	// No SSZ data available, marshal and use JSON
	jsonData, jsonErr := res.marshalJSON()
	if jsonErr != nil {
		return nil, errors.Wrap(jsonErr, "failed to marshal JSON")
	}
	err = c.jsonRestHandler.Post(ctx, endpoint, headers, bytes.NewBuffer(jsonData), nil)
```
I think this makes the code easier to read, especially since the last `err = ...` line is now right next to the error check